pkg/v1/remote/transport/basic.go

```diff
@@ -57,6 +71,19 @@ func (bt *basicTransport) RoundTrip(in *http.Request) (*http.Response, error) {
 				in.Header.Set("Authorization", hdr)
 			}
 		}
+		resp, err := bt.inner.RoundTrip(in)
+		if err != nil {
+			return nil, err
+		}
+
+		if resp.StatusCode == http.StatusUnauthorized || resp.StatusCode == http.StatusForbidden {
+			if last {
```

Comment:
Could we define `last` nearer to where it's used here?

Reply:
BTW, should we rember which `bearerTokenList` work and use a `bearerTransport ` struct member to record it?